lib/modules/dosomething/dosomething_api/resources/kudos_resource.inc

```diff
@@ -145,9 +148,17 @@ function _kudos_resource_access($op) {
   }
 
   if ($op === 'create' || $op === 'delete') {
-    if (!user_is_anonymous()) {
-      return TRUE;
+    // Only auth'd users should be giving Kudos
+    if (user_is_anonymous()) {
+      return FALSE;
+    }
+
+    // We need to make sure people are only creating Kudos for themselves
+    if ($op === 'create' && $user->uid !== $args[1]) {
```

Comment:
@weerd looking at how hard it is to rip user id out. so far its going well... if i hit a major roadblock ill just add the note & merge